cypress/integration/commit_queue.ts

```diff
@@ -0,0 +1,24 @@
+/// <reference types="Cypress" />
+
+const commit_queue = {
+  id: "mongodb-mongo-master",
+};
+const COMMIT_QUEUE_ROUTE = `/commit-queue/${commit_queue.id}`;
+
+describe("commit queue page", function() {
+  beforeEach(() => {
+    cy.server();
+    cy.login();
+  });
+
+  it("Should render the commit queue page with one card", () => {
+    cy.visit(COMMIT_QUEUE_ROUTE);
+    cy.get("[data-cy=commit-queue-card]").should("have.length", 1);
+  });
```

Comment:
these tests will fail if an additional item is added to queue test data. we experienced trouble in the past where a small change to test data created a cascade of test failures because tests and data were tightly coupled. to avoid this we write tests that are robust enough that any data set can be plugged in and tests will still pass. for example, testing that number of cards rendered matches number of cards from response data.

Reply:
That makes sense, thats an interesting approach on university we populated the db with the data we will need for the tests  so we always had the same data/ were able to control what data was used by tests

Reply:
we also use a predefined set of data but we don't think tests should fail if one field on the test data changes. our approach on tests is to make them resilient by matching rendered ui against network responses rather than matching directly on the predefined data.